### PR TITLE
HDDS-9421. Reduce the duplicate create Container/Context code for test code

### PR DESCRIPTION
**Pull request by xichen01**
## What changes were proposed in this pull request?
Reduce the duplicate create Container/Context code for test code

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9421

## How was this patch tested?
Existing test

### CODE REVIEW

**Comment by xichen01** [comment]
All the tests of [e5b39cb](https://github.com/apache/ozone/pull/5429/commits/e5b39cbbdeadafdd6089ef07c5a3943152b19183) have passed : https://github.com/xichen01/ozone/actions/runs/6479407702

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/ContainerTestUtils.java [hunk 1]
```diff
@@ -113,17 +113,23 @@ public static EndpointStateMachine createEndpoint(Configuration conf,
         new LegacyHadoopConfigurationSource(conf));
   }
 
-  public static OzoneContainer getOzoneContainer(
+  public static OzoneContainer getOzoneContainerWithMockContext(
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/ContainerTestUtils.java:116]: I suggest keeping the original name?

* reduce change
* there is no `getOzoneContainer` (without mock context) method we need to differentiate from

Comment (xichen01) [reply hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/ContainerTestUtils.java]: Done.

**Review by adoroszlai — COMMENTED** [review]
Thanks @xichen01 for the patch, LGTM.

I found one more instance of the same mock context:

https://github.com/apache/ozone/blob/76c1747ac6a94693c6e3b170c982b6d6f0c270c8/hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/TestCreatePipelineCommandHandler.java#L87-L90

Please check if it can be replaced.

**Comment by xichen01** [comment]
> Thanks @xichen01 for the patch, LGTM.
> 
> I found one more instance of the same mock context:
> 
> https://github.com/apache/ozone/blob/76c1747ac6a94693c6e3b170c982b6d6f0c270c8/hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/TestCreatePipelineCommandHandler.java#L87-L90
> 
> Please check if it can be replaced.

Thank for your detail check, this has been replaced

**Comment by kerneltime** [comment]
@tanvipenumudy  @Galsza  can you please take a look?

**Review by adoroszlai — APPROVED** [review]
Thanks @xichen01 for updating the patch.

**Pull request merged**